crates/flipperzero/src/furi.rs

```diff
@@ -38,3 +40,108 @@ pub fn sleep(duration: core::time::Duration) {
         }
     }
 }
+
+/// Error codes returned from Furi function calls
+#[derive(Debug)]
+pub enum Error {
+    /// General runtime error, corresponds to FuriStatusError
+    Generic,
+    /// Timeout was exceeded, corresponds to FuriStatusErrorTimeout
+    Timeout,
+    // Resource not available, corresponds to FuriStatusErrorResource
+    Resource,
+    // Parameter error, corresponds to FuriStatusErrorParameter
+    Parameter,
+    // System is out of memory, corresponds to FuriStatusErrorNoMemory
+    NoMemory,
+    // Not allowed in ISR context, corresponds to FuriStatusErrorISR
+    ISR,
+    // Unrecognized error code.
+    Other(u32),
+}
+
+impl From<u32> for Error {
+    fn from(code: u32) -> Self {
+        match i32::from_le_bytes(code.to_le_bytes()) {
```

Comment:
Not sure about this. Surely these a better way to reinterpret a u32 as a i32? (preferably without a transmute or other unsafety?)